### PR TITLE
Add strings.fields() function

### PR DESCRIPTION
**Pull request by alexjx**
introduce fields() function for whitespace splitting.

### CODE REVIEW

**Review by scr-oath — CHANGES_REQUESTED** [review]
Can you add tests to test/test_api.lua? They're structured similar to golang tests

I'm guessing you'll find a bug - i.e. that this method was not added to the `loader.go`'s api

**Comment by alexjx** [comment]
sure, will try adding tests.

I've been using the patch for a while... may the error from creating this patch.

**Comment by scr-oath** [comment]
Here's a sub-PR to your PR https://github.com/alexjx/gopher-lua-libs/pull/1 to show what I had intended - if I comment out the TODO I left for you, the test passes.

**Comment by alexjx** [comment]
Really appreciate your time for reviewing and correcting my PR. I found that the testing structure seems changed since I first start using this lib. I've updated the tests accordingly. Thank you again for helping.

File: strings/example_test.go [hunk 1]
```diff
@@ -25,6 +25,24 @@ func ExampleSplit() {
...
+// strings.fields(string)
+func ExampleFields() {
+	state := lua.NewState()
+	Preload(state)
+	inspect.Preload(state)
+	source := `
+	local strings = require("strings")
+	local inspect = require("inspect")
+	local result = strings.fields("a b c d")
+    print(inspect(result, {newline="", indent=""}))
```
Comment (scr-oath) [inline strings/example_test.go:37]: This appears indented differently from the other lines - I'm on mobile so can't be sure. Can you check?

Comment (alexjx) [reply strings/example_test.go:37]: You're right. The indents are mixed tab with space. I've fixed it.

**Review by scr-oath — APPROVED** [review]

**Pull request merged**